app/models/publication.rb

```diff
@@ -89,11 +89,13 @@ def self.find_or_create_by_sciencewire_id(sw_id)
   end
 
   def self.find_by_doi(doi)
-    PublicationIdentifier.includes(:publication).where(identifier_type: 'doi', identifier_value: doi).map(&:publication)
+    Publication.includes(:publication_identifiers)
+               .find_by("publication_identifiers.identifier_type": 'doi', "publication_identifiers.identifier_value": doi)
   end
 
-  def self.find_by_pmid_in_pub_id_table(pmid)
-    PublicationIdentifier.includes(:publication).where(identifier_type: 'pmid', identifier_value: pmid).map(&:publication)
+  def self.find_by_pmid_pub_id(pmid)
+    Publication.includes(:publication_identifiers)
+               .find_by("publication_identifiers.identifier_type": 'pmid', "publication_identifiers.identifier_value": pmid)
   end
```

Comment:
Why not rename it to `find_by_pmid`?

In general, the revised syntax is not much clearer to read, it's more complex to read IMO.  But, if it works, fine.

Reply:
Correct.  It makes this `find_by` the same as *other* `find_by` methods (which don't require documentation).

Reply:
@darrenleeweber Since `Publication` already has `pmid` field, `find_by_pmid` is a dynamic method that already exists!  This method's usefulness is that it searches `publication_identifiers` instead of `publications` table directly.

Reply:
Oh, it does have a `pmid` field already.  Very weird that this util method even exists then.  Maybe all the callers of this method should be using `Publication.find_by_pmid` instead.  Since it's just one place, might be a good case for removing this method entirely and shove the responsibility for looking up the PublicationIdentifiers in the pubmed_harvester - could make that code a whole lot clearer too.  Can't find any other uses....
```
$ git grep 'find_by_pmid_in_pub_id_table'
app/models/publication.rb:94:  def self.find_by_pmid_in_pub_id_table(pmid)
lib/pubmed_harvester.rb:9:    result = Publication.find_by_pmid_in_pub_id_table(pmid) if result.empty?
```

Reply:
No, the problem is that the data doesn't match up between `publications` and `publication_identifiers` 100% of the time.  This method is intentionally relying on the latter table.

Reply:
I'm open to moving it out, though I don't think it is necessarily part of this PR.

Reply:
Split this concern off into https://github.com/sul-dlss/sul_pub/issues/441